sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -495,53 +494,75 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
-     * set the Content-Length header to zero.</li>
-     * </ul>
+     * @param length Specifies the number of bytes being transmitted in the request body.
      * @return A response that only contains headers and response status code
      * @throws StorageErrorException If you attempt to upload a range that is larger than 4 MB, the service returns
      * status code 413 (Request Entity Too Large)
      */
-    public Mono<Response<FileUploadInfo>> upload(Flux<ByteBuf> data, long length, long offset, FileRangeWriteType type) {
+    public Mono<Response<FileUploadInfo>> upload(Flux<ByteBuf> data, long length, long offset) {
         FileRange range = new FileRange(offset, offset + length - 1);
-        return azureFileStorageClient.files().uploadRangeWithRestResponseAsync(shareName, filePath, range.toString(), type, length, data, null, null, Context.NONE)
+        return azureFileStorageClient.files().uploadRangeWithRestResponseAsync(shareName, filePath, range.toString(), FileRangeWriteType.UPDATE, length, data, null, null, Context.NONE)
             .map(this::uploadResponse);
     }
 
     /**
-     * Uploads file to storage file service.
+     * Clear a range of bytes to specific of a file in storage file service. Clear operations performs an in-place
+     * write on the specified file.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * <p>Upload the file from the source file path. </p>
+     * <p>Clear the range from 0 bytes with length of 1024. </p>
      *
-     * {@codesnippet com.azure.storage.file.fileAsyncClient.uploadFromFile#string}
+     * {@codesnippet com.azure.storage.file.fileAsyncClient.clearRange#long}
      *
      * <p>For more information, see the
-     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-file">Azure Docs Create File</a>
-     * and
-     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/put-range">Azure Docs Upload</a>.</p>
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/put-range">Azure Docs</a>.</p>
      *
-     * @param uploadFilePath The path where store the source file to upload
-     * @return An empty response.
+     * @param length Specifies the number of bytes being cleared in the request body.
+     * @return A response that only contains headers and response status code
+     * @throws StorageErrorException If you attempt to clear a range that is larger than 4 MB, the service returns
+     * status code 413 (Request Entity Too Large)
      */
-    public Mono<Void> uploadFromFile(String uploadFilePath) {
-        return uploadFromFile(uploadFilePath, FileRangeWriteType.UPDATE);
+    public Mono<Response<FileUploadInfo>> clearRange(long length) {
+        FileRange range = new FileRange(0, length - 1);
+        return azureFileStorageClient.files().uploadRangeWithRestResponseAsync(shareName, filePath, range.toString(), FileRangeWriteType.UPDATE, 0, null, null, null, Context.NONE)
```

Comment:
I think the `CLEAR` enum should be used and remove the statement about clearing a range of 4MB, clear is allowed to be as large as the file size. Additionally, clearing will throw an exception is the Content-Length header is set.

Reply:
Done.